File: virtualteachergenaidemo.client/src/models/AgentItem.ts
export class AgentItem {
    id: string;
    name: string;
    description: string;
    prompt: string;
    type: string;
    fileNames: string[];
    features: { feature: string, prompt: string }[] | null | undefined;
    users: { userId: string, sharedUser: boolean }[]; 

    constructor(id: string, name: string, description: string, prompt: string,
        type: string, fileNames: string[],
        features: { feature: string, prompt: string }[] | null | undefined,
        users: { userId: string, sharedUser: boolean }[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.prompt = prompt;
        this.type = type;
        this.fileNames = fileNames;
        this.features = features;
        this.users = users;
    }
}

File: VirtualTeacherGenAIDemo.Server/Models/Storage/AgentItem.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using VirtualTeacherGenAIDemo.Server.Options;
using VirtualTeacherGenAIDemo.Server.Storage;

namespace VirtualTeacherGenAIDemo.Server.Models.Storage
{
    public class AgentItem : IStorageEntity
    {
        [Required, NotEmptyOrWhitespace]
        public string Name { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Description { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Prompt { get; set; } = string.Empty;

        public string Id { get; set; } = string.Empty;

        [Required, NotEmptyOrWhitespace]
        public string Type { get; set; } = string.Empty;

        
        [JsonPropertyName("fileNames")]
        public List<string> FileNames { get; set; } = new List<string>();

        [JsonPropertyName("features")]
        public List<PromptDashboardFeature> Features { get; set; } = new List<PromptDashboardFeature>();

        [JsonIgnore]
        public string Partition => this.Type;

    }

    public class  PromptDashboardFeature
    {
        
        public string Prompt { get; set; } = string.Empty;

        public string Feature { get; set; } = string.Empty;


    }
}
